docs/arch/agent.md

```diff
@@ -0,0 +1,50 @@
+# Summary
+
+The agent is the powerhouse of SealCi. It receives actions and runs them in order to complete the operational part of the CI.
+
+# References
+
+## Actions
+
+A action is a defined as list of commands. actions are launched in a controlled environment, as defined by the agent configuration.
+
+## Artifacts
+
+Artifacts is the result of the execution of a action. We consider as artifacts:
+
+-  logs of execution of the pod
+-  any file that is created during the execution
+
+# Agent operations
+
+## Life of an agent
+
+**Launching & Registering with the Scheduler**  
+Initially, the agent registers with a scheduler. After the schedulers acknowledges the registration, the agent can start processing new actions in order to accept new actions.  
+As part of the registration process, the agent and the scheduler establish a streaming, bi-directional connection.
+
+**Health and Death**  
+An agent streams health and status information to the scheduler, and the agent is kept on the scheduler's resource pool as long as it maintains its connection with it.
+
+**Launching actions**  
+Each time a action is received the agent will:
+
+-  create an container from the configuration of the action
+   -  if the configuration fails the action will considers to be failed
+-  once the configuration is done the list of command will be executed
+   -  for each command, an exit code will be returned to the scheduler
+-  finally, agent will be clean from any remains of the operation
+
+> [!NOTE]  
+> Each time a action stop, artifacts of what happened will be sent back. It can be:
+>
+> -  logs of a fail configuration setup
+> -  logs of a fail execution
+> -  a file created during the execution on succeed
+> -  logs of a succeeded action
+
+## Encapsulated environment
```

Comment:
What is an encapsulated environment?

Reply:
An encapsulated environment is the action's execution environment